include/oneapi/dpl/pstl/zip_view_impl.h

```diff
@@ -0,0 +1,400 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_ZIP_VIEW_IMPL_H
+#define _ONEDPL_ZIP_VIEW_IMPL_H
+
+#if _ONEDPL_CPP20_RANGES_PRESENT
```

Comment:
Did not we think about having `oneapi::dpl::zip_view` as part of namespace `std::ranges` prior to C++23?
It will allow the user to have portable code for each version of the standard. Otherwise, we will need to think about aligning `oneapi::dpl::zip_view` with the C++23 and later as well.

Reply:
Do you mean a kind of name injection into `std::ranges` prior to C++23? Good question. 

On one hand - it is useful when a user is going to modify his code forward to  C++23.
On the other hand, having `std::ranges::zip_view` (in result of name `oneapi::dpl::zip_view` injection), enforces us to be fully compliance with standard `zip_view` C++23. At least, the proposed  zip_view implementation works only with random access views, for example.

I would suggest rather to have an alias `oneapi::dpl::zip_view`, which is our own implementation class prior to C++23 and which is `std::ranges::zip_view` for C++23 and higher.

Reply:
I like the idea of having the alias to `std::ranges::zip_view` in oneDPL for C++23 and higher. It perfectly fits the idea of having everything that should work in the offloaded code in `oneapi::dpl` namespace.